s100py/bag_to_s102.py

```diff
@@ -1,221 +1,76 @@
 import os
 from osgeo import gdal, osr
 import tempfile
-import pprint
 import logging
 
 import numpy
 
-from .s102 import S102File, DEPTH, UNCERTAINTY
-from fuse.raw_read.noaa import bag  # this is from NBS (National Bathymetric Source)
-
+from .s102 import make_s102
 
-def sr_bag_to_s102(input_bag, output_path=""):
-    use_gdal = True
-    use_hdf5 = False
-    use_fuse = True
+# @todo fuse should not be a dependency for reading BAGs.  Use new BAG api?
+from fuse.raw_read.noaa import bag  # this is from NBS (National Bathymetric Source)
 
+def get_valid_epsg() -> list:
+    """
+    Create and return the list of valid EPSG codes for S-102 version 2.0.
+    """
+    valid_epsg = [4326, 5041, 5042]
+    valid_epsg += list(numpy.arange(32601, 32660 + 1))
+    valid_epsg += list(numpy.arange(32701, 32760 + 1))
+    return valid_epsg
+
+def bag_to_s102(input_bag, region, output_path=""):
```

Comment:
To make sure I'm following, You broke the metadata and raster into separate functions and then put them together in this function.  I agree - my intent was to let you guys decide on the best way of getting out of the bag metadata whether that is by fuse or straight HDF5 and builtin xml libs.  

The question that follows is if a dictionary version of the s102 api is desirable.  The way it's built we could extend it to read/write from a dictionary or xml in addition to HDF5 if that somehow is beneficial compared to working with python objects as it does now.